### ISSUE TITLE
[Feature]: Consider adding SQLite support

### ISSUE THREAD

**Issue by oddlama** [immich-app/immich#562]
### Feature detail

I'd like to propose adding SQLite support to Immich. Especially for this use-case, SQLite should be close to the ideal database as opposed to more heavyweight databases like Postgresql and Mysql, at least for personal non-distributed deployments. Refer to [this HN thread and the linked article](https://news.ycombinator.com/item?id=29727707) for a more profound explanation of why it is probably easier and faster for "small-scale" (<100TB) databases.

And thank you for this awesome app!

### Platform

Server

**Comment by bo0tzz** [comment]
We've discussed sqlite before, and while it is a good fit for selfhosted services, it would take a not insignificant effort to move Immich over to it now since we use some postgres-only features for things like search. I still think it would be nice to have in the future, but I believe it is not a priority.

**Comment by webysther** [comment]
The ideia to have multiple services with a central API in this context maybe fix the problem, sqlite for so many years become a great dev tool and for very low number of users or for a single photographer like me, I look as a alternative to photoprism because they start to be the "[emby](https://www.reddit.com/r/selfhosted/comments/13fj282/tutorial_build_your_own_unrestricted_photoprism_ui/)".

I don't see any restriction in the Architecture to use a layer to translate communication with another database.

**Comment by webysther** [comment]
@bo0tzz [full-text-search in sqlite](https://www.sqlite.org/fts5.html) is insane, [calibre last year](https://calibre-ebook.com/whats-new) start support this feature, for my personal library with 10k books took 1hour and the search is perfect with news books added fast. Maybe use only a job to write the fts.db?

**Comment by webysther** [comment]
Based on migration to [prisma](https://github.com/immich-app/immich/pull/1962), please look to [Database features matrix](https://www.prisma.io/docs/reference/database-reference/database-features), for future maybe consider remove any custom commands to support the mostly possible databases will be lovely.

**Comment by avibrazil** [comment]
Me too would like support for MariaDB, SQLite etc.

But now looking at the internals, I don't think this is ever going to happen because Immich uses advanced features of PostgreSQL as vectors, geo computation, vchord for context and face indexing, etc. I can also see many database functions
created by Immich. So PostgreSQL for Immich isn't just tables with data, but has application logic too, in ways that simpler DBs like SQLite and MariaDB can't doo. That would be a lock-in problem in a world of proprietary DBs. But since PostgreSQL and all used extensions are free and open source, it is not. A photo app that can handle machine learning, image recognition, face detection and text search for large photo collections (mine is 1.5TB, 101K photos, 9.5K videos, all assets titled and tagged) needs pretty advanced tools. Tools like PostgreSQL. Not just to hold data, but to process data with modern algorithms.

So, if you are like me and already uses MariaDB for other apps (NextCloud for example), consider migrating these other apps data to PostgreSQL so you can stay with only 1 DB to admin. NextCloud, for example (my case), has a flawless internal tool to migrate data. It is so flawless that in my first migration trial the tool itself already changed the config file to use PostgreSQL and there was zero issues. I'm not going back.

**Comment by grester** [comment]
TL;DR Recommended alternative is PhotoPrism.

For the time being, also because I'm not a fan of setting up such heavy RDBMS for a sole application (given we're talking about containerized environments with multiple "small" apps), the best alternative I've found is PhotoPrism. They advise against SQLite for performance reasons, but you're free to use it anyway. The free version has admin backoffice limited, but all or most features are available anyway by editing the config files or executing commands through the CLI. (For example creating another user is through CLI.)

Another last resort alternative but more quirky is PyGallery2.
